streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -234,15 +234,28 @@ public void shouldRestoreAndNotifyMultipleStores() throws Exception {
         assertThat(callbackTwo.restored.size(), equalTo(3));
 
         assertAllCallbackStatesExecuted(callback, "storeName1");
-        assertCorrectOffsetsReportedByListener(callback, 0L, 10L, 10L);
+        assertCorrectOffsetsReportedByListener(callback, 0L, 9L, 10L);
```

Comment:
should `endOffset` be `9L`, too?

Reply:
actually, this parameters and variables within the verification need to change.  The last parameter is for the number of records restored.  The last offset restored before was incorrect.

Reply:
Ok. That makes sense.

Nevertheless, the test passed comparing `restoreListener.restoreEndOffset, equalTo(endOffset)` -- thus, the `MockStateRestoreListener#restoreEndOffset` is set to `10L` -- this sound wrong.

What I try to say is, that this bug does affect the batch-end-offset and the overall-end-offset. Thus, we should update the Jira description and also add an fix to this PR for the overall-end-offset.